File: android/Reguerta/app/src/main/java/com/reguerta/user/domain/startup/StartupVersionModels.kt
package com.reguerta.user.domain.startup

enum class StartupPlatform(val wireKey: String) {
    ANDROID("android"),
    IOS("ios"),
}

data class StartupVersionPolicy(
    val currentVersion: String,
    val minimumVersion: String,
    val forceUpdate: Boolean,
    val storeUrl: String,
)

sealed interface StartupVersionGateDecision {
    data object Allow : StartupVersionGateDecision

    data class OptionalUpdate(val storeUrl: String) : StartupVersionGateDecision

    data class ForcedUpdate(val storeUrl: String) : StartupVersionGateDecision
}

interface StartupVersionPolicyRepository {
    suspend fun getPolicy(platform: StartupPlatform): StartupVersionPolicy?
}

File: ios/Reguerta/Reguerta/Domain/Startup/StartupVersionModels.swift
import Foundation

enum StartupPlatform: String, Sendable {
    case android
    case ios
}

struct StartupVersionPolicy: Equatable {
    let currentVersion: String
    let minimumVersion: String
    let forceUpdate: Bool
    let storeURL: String
}

enum StartupVersionGateDecision: Equatable, Sendable {
    case allow
    case optionalUpdate(storeURL: String)
    case forcedUpdate(storeURL: String)
}

protocol StartupVersionPolicyRepository: Sendable {
    func policy(for platform: StartupPlatform, environment: SessionEnvironment) async throws -> StartupVersionPolicy
}
